morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -250,6 +259,38 @@ public void ensureCaps() {
     }
 
     @Override
+    public void enableDocumentValidation() {
+        for (final MappedClass mc : mapper.getMappedClasses()) {
```

Comment:
Not in morphia, no.  Classes can be dynamically added/mapped later.  So here, as with indexes and capped collections, classes will have to be explicitly mapped before calling this method.